AP_HAL_ChibiOS:add VUAV-V7pro (#27982)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/VUAV-V7pro/README.md

# VUAV-V7pro Flight Controller

The VUAV-V7pro  flight controller is manufactured and sold by [VIEWPRO](http://www.viewprotech.com/).

## Features

 - STM32H743 microcontroller
 - Three IMUs: ADIS16470,ICM42688,ICM42688
 - Internal vibration isolation for IMUs
 - Internal RM3100 SPI magnetometer
 - Internal two MS5611 SPI barometer
 - Internal RGB LED
 - MicroSD card slot port
 - 1 Analog power port
 - 1 CAN power port
 - 5 UARTs and 1 USB ports
 - 1 RS232 port
 - 14 PWM output ports
 - 4 I2C and 2 CAN ports
 - Safety switch port
 - Buzzer port
 - RC IN port

##  Pinout

![VUAV-V7pro-interface.png](VUAV-V7pro-interface.png)

![VUAV-V7pro-pinout.png](VUAV-V7pro-pinout.png)

## UART Mapping
 - SERIAL0 -> USB

>  - SERIAL1 -> UART2 (Telem1)

Reviewer: ```suggestion
 - SERIAL1 -> UART2 (Telem1) (DMA enabled)

```

Author: Okay, I'll modify it and then force the push

>  - SERIAL2 -> UART6 (Telem2)

Reviewer: ```suggestion
 - SERIAL2 -> UART6 (Telem2)  (DMA enabled)

```

Author: Okay, I'll modify it and then force the push

 - SERIAL3 -> UART1 (GPS1)
 - SERIAL4 -> UART3 (GPS2)

>  - SERIAL5 -> UART8 (USER) TX only

Reviewer: where is this pinned out?the hwdef says its SBUS/DSM but its not named on a connector that I can find

Author: This is on the same pin as RC IN

   <td>GND</td>
   </tr>
   </tbody>
   </table>

### I2C1,I2C2 port

   <table border="1" class="docutils">
   <tbody>
   <tr>
   <th>Pin</th>
   <th>Signal</th>
   <th>Volt</th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>SCL</td>
   <td>+3.3 (pullups)</td>
   </tr>
   <tr>
   <td>3</td>
   <td>SDA</td>
   <td>+3.3 (pullups)</td>
   </tr>
   <tr>
   <td>4</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   </tbody>
   </table>

### USB Ex

   <table border="1" class="docutils">
   <tbody>
   <tr>
   <th>Pin </th>
   <th>Signal </th>
   <th>Volt </th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC IN</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>D_minus</td>
   <td>+3.3V</td>
   </tr>
   <tr>
   <td>3</td>
   <td>D_plus</td>
   <td>+3.3V</td>
   </tr>
   <tr>
   <td>4</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   </tbody>
   </table>

### RSSI

   <table border="1" class="docutils">
   <tbody>
   <tr>
   <th>Pin</th>
   <th>Signal</th>
   <th>Volt</th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>RSSI</td>
   <td>up to +3.3V</td>
   </tr>
   <tr>
   <td>3</td>

>    <td>SBUS_OUT</td>

Reviewer: what uart pin is this, and no uarts are setup to provide SBUS out protocol in the hwdef

Author: This SBUS, DSM as the same port，Can be used as a universal serial port output

Reviewer: It should be named UART8 TX unless you default UART8 protocol to "15" for sbus servo output....and if you really have UART8 RX tied the the same pin as RCIN, we should note that in the UART section....in which case you could set UART8 protocol to 23 (RCIN) and use its TX pin (currently mislabeled as SBUS_OUT) anf the RCIN pin for CRSF/ELRS,etc.

   <th>Pin</th>
   <th>Signal</th>
   <th>Volt</th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>ADC_3V3</td>
   <td>up to +3.3V</td>
   </tr>
   <tr>
   <td>3</td>
   <td>ADC_6V6</td>
   <td>up to +6.6V</td>
   </tr>
   <tr>
   <td>4</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   </tbody>
   </table>

### POWER1

   <table border="1" class="docutils">
   <tbody>
   <tr>
   <th>Pin</th>
   <th>Signal</th>
   <th>Volt</th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC IN</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>VCC IN</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>3</td>
   <td>CURRENT</td>
   <td>up to +3.3V</td>
   </tr>
   <tr>
   <td>4</td>
   <td>VOLTAGE</td>
   <td>up to +3.3V</td>
   </tr>
   <td>5</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   <td>6</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   </tbody>
   </table>

### POWER2

   <table border="1" class="docutils">
   <tbody>
   <tr>
   <th>Pin</th>
   <th>Signal</th>
   <th>Volt</th>
   </tr>
   <tr>
   <td>1</td>
   <td>VCC IN</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>2</td>
   <td>VCC IN</td>
   <td>+5V</td>
   </tr>
   <tr>
   <td>3</td>

>    <td>CAN_H</td>

Reviewer: which can bus is used?should be labelled

Author: The chip uses MCP2542FD-E/MF,Should it be reflected above?

Reviewer: not necessary.....just users need to know what can bus number belongs to a given can bus pinout, because some can devices can not co-exist on the same bus

Author: Okay, I will mark whether it is CAN1 or CAN2

Reviewer: still needs the pin label updated

   <td>+24V</td>
   </tr>
   <tr>
   <td>4</td>
   <td>CAN_L</td>
   <td>+24V</td>
   </tr>
   <td>5</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   <td>6</td>
   <td>GND</td>
   <td>GND</td>
   </tr>
   </tbody>
   </table>

## RC Input

RC input is configured on the RCIN pin, at one end of the servo rail. This pin supports all unidirectional RC protocols.

## PWM Output

> The VUAV-V7pro supports up to 14 PWM outputs,support all PWM protocols as well as DShot. All 14 PWM outputs have GND on the top row, 5V on the middle row and signal on the bottom row.

Reviewer: not correct....
```suggestion
The VUAV-V7pro supports up to 14 PWM outputs,support all PWM protocols. Outputs 1-12 support  DShot. All 14 PWM outputs have GND on the top row, 5V on the middle row and signal on the bottom row.

```

Author: OK

The 14 PWM outputs are in 4 groups:

 - PWM 1, 2, 3 and 4 in group1
 - PWM 5, 6, 7 and 8 in group2
 - PWM 9, 10, 11 and 12 in group3
 - PWM 13, 14 in group4

> Channels 1-8 support bi-directional Dshot.

Reviewer: Move inot first sentence where output capabilities are discussed.

Author: OK

   <td>PWM1</td>
   <td>50</td>
   <td>PWM8</td>
   <td>57</td>
   </tr>
   <tr>
   <td>PWM2</td>
   <td>51</td>
   <td>PWM9</td>
   <td>58</td>
   </tr>
   <tr>
   <td>PWM3</td>
   <td>52</td>
   <td>PWM10</td>
   <td>59</td>
   </tr>
   <tr>
   <td>PWM4</td>
   <td>53</td>
   <td>PWM11</td>
   <td>60</td>
   </tr>
   <tr>
   <td>PWM5</td>
   <td>54</td>
   <td>PWM12</td>
   <td>61</td>
   </tr>
   <tr>
   <td>PWM6</td>
   <td>55</td>
   <td>PWM13</td>
   <td>62</td>
   </tr>
   <tr>
   <td>PWM7</td>
   <td>56</td>
   <td>PWM14</td>
   <td>63</td>
   </tr>
   </tbody>
   </table>

## Analog inputs

The VUAV-V7pro flight controller has 5 Analog inputs

 - ADC Pin2-> Battery Current
 - ADC Pin16 -> Battery Voltage 
 - ADC Pin19 -> ADC 3V3 Sense
 - ADC Pin3 -> ADC 6V6 Sense
 - ADC Pin9 -> RSSI voltage monitoring

## Battery Monitor Configuration

The board has voltage and current inputs sensor on the POWER1 ADC and POWER2 CAN connector.

The correct battery setting parameters are:

Enable Battery1 monitor:
 - BATT_MONITOR   4
 - BATT_VOLT_MULT 15.7 (may need adjustment if supplied monitor is not used)
 - BATT_AMP_PERVLT 60.61 (may need adjustment if supplied monitor is not used)

> 

Reviewer: ```suggestion

- BATT_VOLT_PIN 16
- BATT_CUR_PIN 2
```

Author: I previously considered that the default PIN would be this value, so I removed these two parameters here. I am now adding them again.
